p2pp/gcode.py

```diff
@@ -70,23 +70,30 @@ def __init__(self, gcode_line):
 
                 fields = fields[1:]
 
-        self.X = self.get_parameter("X", None)
-        self.Y = self.get_parameter("Y", None)
-        self.Z = self.get_parameter("Z", None)
-        self.E = self.get_parameter("E", None)
+                self.X = self.get_parameter("X", None)
+                self.Y = self.get_parameter("Y", None)
+                self.Z = self.get_parameter("Z", None)
+                self.E = self.get_parameter("E", None)
 
     def __str__(self):
         p = ""
-        for key in self.Parameters:
-            form = "{}{} "
-            if self.Command in ["G0", "G1"]:
+
+        # use the same formatting as prusa to ease file compares (X, Y, Z, E, F)
+        sorted_keys = "XYZEF"
+        for key in sorted_keys:
+            if self.fullcommand in ['G0', 'G1'] and key in self.Parameters:
+                form = ""
                 if key in "XYZ":
                     form = "{}{:.3f} "
                 if key in "E":
-                    form = "{}{:.4f} "
+                    form = "{}{:.5f} "
```

Comment:
We accumulate a fair bit of rounding error with .4, most prusa generated gcode is using 5 decimals (though not all).